toolchains/solidity/core/crates/slither-server/src/types.rs

```diff
@@ -1,6 +1,49 @@
 use serde::{Deserialize, Serialize};
 use tower_lsp::lsp_types::{Diagnostic, DiagnosticSeverity as Severity, Position, Range};
+use std::vec;
+use tokio::sync::mpsc::{Receiver, Sender};
+use tokio_util::sync::CancellationToken;
+use tower_lsp::lsp_types::*;
 
+#[derive(Debug)]
+pub struct SlitherDiag {
+    pub diagnostics: Vec<Diagnostic>,
+    pub uri: Url,
+}
+
+impl SlitherDiag {
+    pub fn new(uri: Url, diagnostics: Vec<Diagnostic>) -> Self {
+        Self { uri, diagnostics }
+    }
+}
+
+#[derive(Debug)]
+pub struct SlitherData {
+    pub slither_processes: Vec<CancellationToken>,
+    pub receiver: Option<Receiver<SlitherDiag>>,
+    pub sender: Sender<SlitherDiag>,
+    pub libs_paths: Vec<String>,
+    pub src_paths: Vec<String>,
+    pub tests_paths: Vec<String>,
```

Comment:
shouldn't be vectors

Reply:
It should, since in documentation of foundry.toml the aliases can be arrays of alliases

Reply:
oh ok mb didn't knew